airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -594,12 +594,16 @@ def _handle_error_state(self, hook: DataprocHook, cluster: Cluster) -> None:
         if cluster.status.state != cluster.status.State.ERROR:
             return
         self.log.info("Cluster is in ERROR state")
+        self.log.info("Gathering diagnostic information.")
         gcs_uri = hook.diagnose_cluster(
             region=self.region, cluster_name=self.cluster_name, project_id=self.project_id
         )
         self.log.info("Diagnostic information for cluster %s available at: %s", self.cluster_name, gcs_uri)
         if self.delete_on_error:
             self._delete_cluster(hook)
+            # The delete op is asynchronous and can cause further failure if the cluster finishes
+            # deleting between catching AlreadyExists and checking state
+            self._wait_for_cluster_in_deleting_state(hook)
             raise AirflowException("Cluster was created but was in ERROR state.")
         raise AirflowException("Cluster was created but is in ERROR state")
```

Comment:
```suggestion
        raise AirflowException("Cluster was created but is in ERROR state")
```

I know its not a part of this PR, but can we remove this inner Exception? I think its redundant with the one below it, which would be raised anyways if this one were removed.

Reply:
I'll remove it.

Reply:
Actually, how about we keep it and I'll add a suffix that the cluster was deleted. That was possibly the intent of the original message and it was forgotten. 

`_wait_for_cluster_in_deleting_state` Doesn't state the completion of cluster deletion anyway.